core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -161,6 +161,20 @@ static long offsetForLatestTimestamp(TopicIdPartition topicIdPartition, ReplicaM
         return timestampAndOffset.get().offset;
     }
 
+    /**
+     * The method is used to get the offset for the given timestamp for the topic-partition.
+     *
+     * @return The offset for the given timestamp.
+     */
+    static long offsetForTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager, long timestampToSearch, int leaderEpoch) {
+        Option<FileRecords.TimestampAndOffset> timestampAndOffset = replicaManager.fetchOffsetForTimestamp(
+            topicIdPartition.topicPartition(), timestampToSearch, Option.empty(), Optional.of(leaderEpoch), true).timestampAndOffsetOpt();
+        if (timestampAndOffset.isEmpty()) {
+            throw new OffsetNotAvailableException("Offset for timestamp: " + timestampToSearch + " not found for topic partition: " + topicIdPartition);
+        }
+        return timestampAndOffset.get().offset;
+    }
+
```

Comment:
Should we send the the isolation level as well instead of `Option.empty()`? Also we can have common method and send respective timestamps from: `offsetForEarliestTimestamp`, `offsetForLatestTimestamp` and `offsetForTimestamp` methods.